tuning/sft_trainer.py

```diff
@@ -155,6 +155,23 @@ def train(
                 "Trainer should not perform packing when using `--padding_free`"
             )
 
+    if fast_moe_config is not None:
+        # Checking for unsupported modules with Scatter MoE for LoRA
+        restricted_modules = ["all-linear", "output_linear", "input_linear", "router"]
```

Comment:
Lets not hard code `"output_linear", "input_linear", "router"` since these vary model to model. I understand this is the best attempt, however, lets not do this. Rather, lets only restrict `all-linear` however, raise a warning that modules that are part of moe model if specific can lead to unexpected behaviour or unsuccessful tuning. And then place the recommendation as well saying restrict to linear layers outside moe module like the attention layer (q_proj, k_proj ..) etch. WDYT?